### PR TITLE
[hue] update action for July 2024

### PR DESCRIPTION
**Pull request by cjac**
modified:   hue/README.md

* line-wrapped and clarified important note

modified:   hue/hue.sh

* added functions os_id, is_debian, is_debian10, is_ubuntu
* only adding buster-backports to sources.list for debian10
* update_repo now uses is_debian || is_ubuntu rather than checking for apt-get in path
* setting DEBIAN_FRONTEND=noninteractive before installation of hue ; mysql-community-server had an important message that it was prompting the user to answer on a non-interactive console, halting tests.
* added note that we may need to specify this option with debconf

* jettisoned my function remove_old_backports with preference for Prince's much better replace_backports_repo

* using slightly different logic to determine whether we replace the backports repo

### CODE REVIEW

**Comment by cjac** [comment]
The previous run didn't seem to get the message that the installation was noninteractive and stopped again with the prompt.  I do not see this behavior in my reproduction environment.

**Comment by cjac** [comment]
last test revealed that I had failed to define os_version on which is_debian depends.  This should fix that last failure.

**Comment by cjac** [comment]
@prince-cs - this PR is ready for squash and merge.  I believe i granted you authorization to perform reviews.  Please review and approve if you believe we're ready to push this one out.  I know that #1192 is blocked on hue and probably others.

**Review by prince-cs — APPROVED** [review]

**Pull request merged**
